spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTMPeephole.scala

```diff
@@ -76,11 +76,9 @@ class LSTMPeephole[T : ClassTag] (
         .add(SelectTable(1))
         .add(NarrowTable(2, 2)))
 
-  override def preTopology: AbstractModule[Activity, Activity, T] =
-    Sequential()
-    .add(Dropout(p))
-    .add(TimeDistributed(Linear(inputSize, hiddenSize * 4, wRegularizer = wRegularizer,
-      bRegularizer = bRegularizer)))
+  override var preTopology: TensorModule[T] =
+    Linear(inputSize, hiddenSize * 4, wRegularizer = wRegularizer,
```

Comment:
if p is non zero, the pre-topology should be null (just as like LSTM)

Reply:
I am worried if we set pre-topology is null, there would be a perf downgrade(LSTM didn't have this problem because its implementation is different). Change here is moving dropout from before linear to after linear, like update dropout -> linear to linear -> dropout. They may be not exactly the same, but both of them randomly drop values which will be feeded to next layer.